apps/server/src/shared/repo/dashboard/dashboard.model.mapper.ts

```diff
@@ -52,22 +52,35 @@ export class DashboardModelMapper {
 		throw new InternalServerErrorException('unknown learnroom type');
 	}
 
-	private async instantiateGridElementModel(gridElement: IGridElement): Promise<DashboardGridElementModel> {
-		if (!gridElement.hasId()) {
-			return new DashboardGridElementModel();
+	async mapGridElementToModel(
+		elementWithPosition: GridElementWithPosition,
+		dashboard: DashboardModelEntity
+	): Promise<DashboardGridElementModel> {
+		const { gridElement } = elementWithPosition;
+		if (gridElement.hasId()) {
+			const existing = await this.em.findOne(DashboardGridElementModel, gridElement.getId() as string);
+			if (existing) return this.updateExistingGridElement(existing, elementWithPosition, dashboard);
```

Comment:
what happens when hasId is true while existing is null?
beside, not return an action/method call, always do a call to fetch data first, then return data (code style)
```suggestion
			if (existing) {
					const updated =	 this.updateExistingGridElement(existing, elementWithPosition, dashboard);
					return updated;
			}
```

Reply:
note to self, extract the two ifs